### DISCUSSION TITLE
Plugin "cycling"

### DISCUSSION THREAD

**Discussion by samdafi** [apohl79/audiogridder#1210] [Q&A] [answered]
I have a problem with a plugin under development where it is constantly "cycling" on Mac M1 in the AudioGridder server (open, close, open, close, etc). This happens in all options for "sandbox mode." This also happens when I run the server under lldb to attach to the server process to tell me what could be wrong with the plugin, which is operating well on pretty much all other hosts. 

None of the logs as documented on the log locations page indicate a crash, though that's certainly what it looks like! Any pointers on how to debug this, or some way I can get the AudioGridder server into a state where it will provide a crash log with a trace, without having to build it myself? Is there a secondary process the server launches that I should be trying to attach to with lldb, instead of the server itself?

Thanks!

-Sam

**Comment by apohl79** [comment, marked as answer]
Hey Sam,

the best way to debug this is turning off sandboxing and running the server under lldb with the "-server" commandline flag. This will run a server process directly which loads the plugin directly too. You will be able to break into your plugin that way. It might very much also be an issue in the server, so you might want to compile it yourself so you get better insights of whats going on.

>  Is there a secondary process the server launches that I should be trying to attach to with lldb, instead of the server itself?

The server process is running sub processes in certain situations. Sandboxes of course, but also when you run the server w/o parameters, it runs as a master process which starts a server sub process (thats why you should use the "-server" flag in lldb). This is for auto restarting a crashed server process, which can easily happen w/o sandboxing.

Best,
Andreas

**Comment by samdafi** [comment]
Hello Andreas,

Thank you so much! This project is really cool and I would like my users to be able to drop my products in it without worry. I will give your suggestions a shot and try to remember to write down my debugging steps here in case they help anybody else in the future, before closing the discussion.

-Sam

**Reply by samdafi** [reply]
Steps laid out, as described by Andreas, on Mac

- lldb AudioGridderServer.app/Contents/MacOS/AudioGridderServer
- process launch -- -server
- open the audio gridder preferences and make sure sandbox is disabled (remember to apply)
- open audio gridder plugin on client
- make your plugin crash

Looks like in my case, I had an asynchronous callback on the message thread that happened after the editor had been deleted, so perhaps it's opened twice or there is a detail about VST3 that I must fix. In any case, I made the callback safer.

**Reply by apohl79** [reply]
thanks for this!
